src/Microsoft.TemplateEngine.Cli/Commands/NewCommand.Help.cs

```diff
@@ -19,11 +21,9 @@ public IEnumerable<HelpSectionDelegate> CustomHelpLayout()
                     throw new ArgumentException($"{nameof(context)} should be for {nameof(NewCommand)}");
                 }
                 NewCommandArgs args = new NewCommandArgs(newCommand, context.ParseResult);
-                InstantiateCommand instantiateCommand = InstantiateCommand.FromNewCommand(newCommand);
-
-                //tokens do not contain help option
-                ParseResult reparseResult = ParserFactory.CreateParser(instantiateCommand).Parse(args.Tokens);
-                instantiateCommand.WriteHelp(context, reparseResult);
```

Comment:
@baronfel the bug that I mentioned was here (and in similar parts below).
We used to create `InstantiateCommand` from `NewCommand` before for instantiation using `dotnet new console`.
This is logically not correct as you are modifying the command (which is not possible). This way it worked but lost all connection to initial command.

I changed it not to do it, instead just use (static now) handlers of `InstantiateCommand` also in `NewCommand` and create `InstantiateCommandArgs` from `NewCommandArgs`. This fixed it, also I was able to remove many hacks all over.